File: bigbear/615/35.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        if(nums.length == 0) return 0;
        // if(target > nums[nums.length - 1]) return nums.length;
        
        int lo = 0, hi = nums.length - 1;
        while(lo <= hi){
            int mid = (hi - lo) / 2 + lo;
            if(nums[mid] == target) return mid;
            else if(nums[mid] > target) hi = mid - 1;
            else lo = mid + 1;
        }
        return lo;
    }
}

File: zemiao/35.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        left, right = 0, len(nums)-1
        while left < right:
            mid = left + (right-left)/2
            if nums[mid] < target:
                left = mid +1
            else:
                right = mid -1
        
        return left if target<=nums[left] else left+1